src/main/java/seedu/address/model/person/Parent.java

```diff
@@ -0,0 +1,88 @@
+package seedu.address.model.person;
+
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a parent in the address book
+ */
+public class Parent extends Person {
+    private Student child = null;
+
+    /**
+     * Constructs a {@code Parent} with the given details.
+     */
+    public Parent(Name name, Phone phone, Email email, Address address, Student child, Set<Tag> tags) {
+        super(name, phone, email, address, tags);
+        this.child = child;
+    }
+
+    /**
+     * Constructs a {@code Parent} with the given {@code Person} as a base.
+     */
+    public Parent(Person person, Student child) {
+        super(person.getName(), person.getPhone(), person.getEmail(), person.getAddress(), person.getTags());
+        this.child = child;
+    }
+
+    public void setChild(Student child) {
+        this.child = child;
+    }
+
+    public Student getChild() {
+        return child;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof Parent)) {
+            return false;
+        }
+
+        Parent otherParent = (Parent) other;
+        return super.equals(otherParent) && getChild().equals(otherParent.getChild());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.getName(), this.getPhone(), this.getEmail(), this.getAddress(),
+                this.getChild(), this.getTags());
```

Comment:
Perhaps you could consider adding a test for the hashCode method? You could create a test case in which you obtain the individual fields manually, compute the hash and use assertEquals() to compare the resultant value with hashCode()